common/lib/types/pushchannelsubscription.js

```diff
@@ -0,0 +1,63 @@
+var PushChannelSubscription = (function() {
+	var msgpack = (typeof require !== 'function') ? Ably.msgpack : require('msgpack-js');
+
+	function PushChannelSubscription() {
+		this.channel = undefined;
+		this.deviceId = undefined;
+		this.clientId = undefined;
+	}
+
+	/**
+	 * Overload toJSON() to intercept JSON.stringify()
+	 * @return {*}
+	 */
+	PushChannelSubscription.prototype.toJSON = function() {
+		return {
+			channel: this.channel,
+			deviceId: this.deviceId,
+			clientId: this.clientId,
+		};
+	};
+
+	PushChannelSubscription.prototype.toString = function() {
+		var result = '[PushChannelSubscription';
+		if(this.channel)
+			result += '; channel=' + this.channel;
+		if(this.deviceId)
+			result += '; deviceId=' + this.deviceId;
+		if(this.clientId)
+			result += '; clientId=' + this.clientId;
+		result += ']';
+		return result;
+	};
+
+	PushChannelSubscription.toRequestBody = function(subscription, format) {
+		return (format == 'msgpack') ? msgpack.encode(subscription, true): JSON.stringify(subscription);
+	};
+
+	PushChannelSubscription.fromResponseBody = function(body, format) {
+		if(format)
+			body = (format == 'msgpack') ? msgpack.decode(body) : JSON.parse(String(body));
+
+		for(var i = 0; i < body.length; i++) {
+			body[i] = PushChannelSubscription.fromDecoded(body[i]);
+		}
+		return body;
```

Comment:
Could this not be replaced with `return PushChannelSubscription.fromValuesArray(body)`, and get rid of `PushChannelSubscription.fromDecoded`?